src/pages/Vote/VotePage.tsx

```diff
@@ -312,8 +341,32 @@ export default function VotePage({
                 <Trans>Vote Against</Trans>
               </ButtonPrimary>
             </RowFixed>
-          ) : (
-            ''
+          )}
+          {showQueueButton && (
+            <RowFixed style={{ width: '100%', gap: '12px' }}>
+              <ButtonPrimary
+                padding="8px"
+                $borderRadius="8px"
+                onClick={() => {
+                  toggleQueueModal()
+                }}
+              >
+                <Trans>Queue</Trans>
+              </ButtonPrimary>
+            </RowFixed>
+          )}
+          {showExecuteButton && (
+            <RowFixed style={{ width: '100%', gap: '12px' }}>
+              <ButtonPrimary
+                padding="8px"
+                $borderRadius="8px"
+                onClick={() => {
+                  toggleExecuteModal()
+                }}
+              >
+                <Trans>Execute</Trans>
```

Comment:
Consider moving the time-based logic here (but still only showing it if `currentTimestamp && proposalData?.eta`.

To the button, add:
```
disabled={currentTimestamp.lt(proposalData.eta)}
```

The the text, add something like:
```
<Trans>Execute (after {toLocaleDate(proposalData.eta)})</Trans>
```